src/ophys_etl/modules/segmentation/postprocess_utils/roi_merging.py

```diff
@@ -0,0 +1,999 @@
+                row.append(False)
+        mask.append(row)
+
+    new_roi = ExtractROI(x=roi.x0,
+                         y=roi.y0,
+                         width=roi.width,
+                         height=roi.height,
+                         mask=mask,
+                         valid_roi=True,
+                         id=roi.roi_id)
+    return new_roi
+
+
+def _do_rois_abut(array_0: np.ndarray,
+                  array_1: np.ndarray,
+                  dpix: float = np.sqrt(2)) -> bool:
+    """
+    Method that does the work behind user-facing do_rois_abut.
+
+    This method takes in two arrays of pixel coordinates
+    calculates the distance between every pair of pixels
+    across the two arrays. If the minimum distance is less
+    than or equal dpix, it returns True. If not, it return False.
+
+    Parameters
+    ----------
+    array_0: np.ndarray
+        Array of the first set of pixels. Shape is (npix0, 2).
+        array_0[:, 0] are the row coodinates of the pixels
+        in array_0. array_0[:, 1] are the column coordinates.
+
+    array_1: np.ndarray
+        Same as array_0 for the second array of pixels
+
+    dpix: float
+        Maximum distance two arrays can be from each other
+        at their closest point and still be considered
+        to abut (default: sqrt(2)).
+
+    Return
+    ------
+    boolean
+    """
+    distances = cdist(array_0, array_1, metric='euclidean')
+    if distances.min() <= dpix:
+        return True
+    return False
+
+
+def _get_pixel_array(roi: OphysROI) -> np.ndarray:
+    """
+    get Nx2 array of pixels (in global coordinates)
+    that are in the ROI
+
+    Parameters
+    ----------
+    OphysROI
+
+    Returns
+    -------
+    np.ndarray
+    """
+    mask = roi.mask_matrix
+    n_bdry = mask.sum()
+    roi_array = -1*np.ones((n_bdry, 2), dtype=int)
+    i_pix = 0
+    for ir in range(roi.height):
+        row = ir+roi.y0
+        for ic in range(roi.width):
+            col = ic+roi.x0
+            if not mask[ir, ic]:
+                continue
+
+            roi_array[i_pix, 0] = row
+            roi_array[i_pix, 1] = col
+            i_pix += 1
+
+    if roi_array.min() < 0:
+        raise RuntimeError("did not assign all boundary pixels")
+
+    return roi_array
+
+
+def do_rois_abut(roi0: OphysROI,
+                 roi1: OphysROI,
+                 dpix: float = np.sqrt(2)) -> bool:
+    """
+    Returns True if ROIs are within dpix of each other at any point.
+
+    Parameters
+    ----------
+    roi0: OphysROI
+
+    roi1: OphysROI
+
+    dpix: float
+        The maximum distance from each other the ROIs can be at
+        their closest point and still be considered to abut.
+        (Default: np.sqrt(2))
+
+    Returns
+    -------
+    boolean
+
+    Notes
+    -----
+    dpix is such that if two boundaries are next to each other,
+    that corresponds to dpix=1; dpix=2 corresponds to 1 blank pixel
+    between ROIs
+    """
+    array_0 = _get_pixel_array(roi0)
```

Comment:
the image-based way might save some space, maybe not:
```
touching = np.count_nonzero(roi1.mask, binary_dilation(roi2.mask)` != 0
```
https://scikit-image.org/docs/stable/api/skimage.morphology.html?highlight=binary_dilation#skimage.morphology.binary_dilation